### ISSUE TITLE
No communication between provider and consumer

### ISSUE THREAD

**Issue by andreisanta09** [bmwcarit/joynr#27]
Hi,

I want to build a provider (Java Spring Boot) / consumer (Java JSF) pair using Joynr Java but I'm running into an issue when messages are sent from one to the other.

I am working on a Win 10 machine, Payara 4.1, Mosquitto MQTT and the needed joynr infrastructure. I already have a running provider / consumer as Java console apps (Joynr 0.25.2) which work fine.

I tried out your example (radio-app) from the last releases but none seem to work for me:
Steps:
- Maven install in the radio-app folder
- start the database / domain 
- deploy the joynr backend services
- MQTT broker is up-and-running
- start the provider app using my local domain as parameter
- start the consumer app using my local domain as parameter

**Tag 1.3.2**:
- both applications start, consumer crashes after a few seconds : 
_[ERROR] MyRadioConsumerApplication: No provider found
io.joynr.exceptions.DiscoveryException: Arbitration and Connector failed: domain: [joynr-infrastructure] interface: vehicle/Radio qos: io.joynr.arbitration.DiscoveryQos@15e3929: Arbitration could not be finished in time._
- I am able to trigger a broadcast or a shuffling on the provider side via command line

**Tag 1.4.0** : 
- both applications start, consumer crashes after a few seconds : 
_[ERROR] MyRadioConsumerApplication: No provider found
io.joynr.exceptions.DiscoveryException: Arbitration and Connector failed: domain: [joynr-infrastructure] interface: vehicle/Radio qos: io.joynr.arbitration.DiscoveryQos@15e3929: Arbitration could not be finished in time._
- no command line input is accepted by the provider (e.g. 'w' or 's')

For both these releases I can see all the messages coming from the provider or the consumer in the MQTT broker (using MQTT Lens) but is seems the messages don't get across to the other app (similar with what I see on the applications written by me). 

Any ideas / hints on what I am missing ?

Thanks,
Andrei

**Comment by manuel-schiller** [comment]
How exactly do you start the provider & consumer processes (command line arguments)?
Can you please provide detailed logs (log level DEBUG for io.joynr) for both consumer and provider process as well as the mqtt broker?

**Comment by manuel-schiller** [comment]
Registration of your provider in global capabilities directory failed, hence it cannot be found by the consumer:

    2018-08-23 16:12:44,810 [INFO ] LocalCapabilitiesDirectoryImpl: global registration for 70c7e1b1-210b-4f3f-8062-709c06abf025, joynr-infrastructure : vehicle/Radio failed
    2018-08-23 16:12:44,810 [DEBUG] RequestInterpreter: execute request on provider onRejection: joynr.exceptions.ProviderRuntimeException: io.joynr.exceptions.JoynrTimeoutException: ttl expired on: 23/08 16:12:44:044, request: Request: add, requestReplyId: 6b93efa9-2413-46fb-b65b-03d84f52f9fb, params: [DiscoveryEntry [providerVersion=Version [majorVersion=0, minorVersion=0], domain=joynr-infrastructure, interfaceName=vehicle/Radio, participantId=70c7e1b1-210b-4f3f-8062-709c06abf025, qos=ProviderQos [customParameters=[], priority=1535029904723, scope=GLOBAL, supportsOnChangeSubscriptions=false], lastSeenDateMs=1535029904738, expiryDateMs=1538658704738, publicKeyId=], true]
    2018-08-23 16:12:44,810 [ERROR] DispatcherImpl: Error processing request: Request: add, requestReplyId: 6b93efa9-2413-46fb-b65b-03d84f52f9fb, params: [DiscoveryEntry [providerVersion=Version [majorVersion=0, minorVersion=0], domain=joynr-infrastructure, interfaceName=vehicle/Radio, participantId=70c7e1b1-210b-4f3f-8062-709c06abf025, qos=ProviderQos [customParameters=[], priority=1535029904723, scope=GLOBAL, supportsOnChangeSubscriptions=false], lastSeenDateMs=1535029904738, expiryDateMs=1538658704738, publicKeyId=], true]
    joynr.exceptions.ProviderRuntimeException: io.joynr.exceptions.JoynrTimeoutException: ttl expired on: 23/08 16:12:44:044

You wrote that you "deploy the joynr backend services".
What exactly are you doing here?
Please also attach logs from the global capabilities directory.

**Comment by andreisanta09** [comment]
Hi,

I deployed the backend services on the payara server using the commands in the tutorial : 
_bin/asadmin deploy <RADIO_HOME>/target/discovery-jee.war
bin/asadmin deploy <RADIO_HOME>/target/accesscontrol-jee.war_

I deleted the existing services and re-deployed them and everything works as expected - not sure what happened with the initial services. 
Even the applications written by me are now communicating between each other. From my point of view this issue can be closed.

Hope I didn't waste to much of your time with this trivial issue.

Thanks for the support,
Andrei
